plugins/completion/codeium-nvim.nix

```diff
@@ -0,0 +1,121 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "codeium-nvim";
+  originalName = "codeium.nvim";
+  luaName = "codeium";
+
+  maintainers = with lib.maintainers; [
+    GaetanLepage
+    khaneliman
+  ];
+
+  # TODO: added 2024-09-03 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "configPath"
+    "binPath"
+    [
+      "api"
+      "host"
+    ]
+    [
+      "api"
+      "port"
+    ]
+    [
+      "tools"
+      "uname"
+    ]
+    [
+      "tools"
+      "uuidgen"
+    ]
+    [
+      "tools"
+      "curl"
+    ]
+    [
+      "tools"
+      "gzip"
+    ]
+    [
+      "tools"
+      "languageServer"
+    ]
+    "wrapper"
+  ];
+
+  # Register nvim-cmp association
+  imports = [
+    { cmpSourcePlugins.codeium = "codeium-nvim"; }
+  ];
+
+  settingsOptions = {
+    config_path = defaultNullOpts.mkStr {
+      __raw = "vim.fn.stdpath('cache') .. '/codeium/config.json'";
+    } "The path to the config file, used to store the API key.";
+
+    bin_path = defaultNullOpts.mkStr {
+      __raw = "vim.fn.stdpath('cache') .. '/codeium/bin'";
+    } "The path to the directory where the Codeium server will be downloaded to.";
+
+    api = {
+      host = defaultNullOpts.mkStr "server.codeium.com" ''
+        The hostname of the API server to use.
+      '';
+
+      port = defaultNullOpts.mkNullableWithRaw' {
+        type = with lib.types; lib.nixvim.transitionType ints.positive toString (strMatching "[0-9]+");
```

Comment:
```suggestion
        # TODO: Added 2024-09-05; remove after 24.11
        type = with lib.types; lib.nixvim.transitionType ints.positive toString (strMatching "[0-9]+");
```

Reply:
Also, is it `lib.nixvim.transitionType` or `lib.types.transitionType`? I guess if tests pass, this _is_ correct.